src/webnn_native/Operand.cpp

```diff
@@ -21,8 +21,14 @@
 
 namespace webnn_native {
 
-    OperandBase::OperandBase(GraphBuilderBase* graphBuilder, std::vector<Ref<OperandBase>> inputs)
-        : ObjectBase(graphBuilder->GetContext()), mInputs(std::move(inputs)) {
+    OperandBase::OperandBase(GraphBuilderBase* graphBuilder, OperatorBase* operatorBase)
+        : ObjectBase(graphBuilder->GetContext()), mOperator(operatorBase) {
+        if (mOperator->Inputs().size() >= 1) {
+            auto primaryInput = mOperator->Inputs()[0];
+            // The type and rank is the same as input[0] by default.
+            mType = primaryInput->Type();
+            mRank = primaryInput->Rank();
```

Comment:
Would the default type and rank be necessary?

Reply:
Type can be float32 by default, is it 0 for the rank?